BlueBrain/Search
# Upgrade to dvc>=2  + release bluesearch v0.1.1

Fixes #273.

## Description

Upgrades to `dvc==2.0.6` and releases `bluesearch v0.1.1`.

### Notes
1. I used to consistently get the error 
```ValueError: unsupported pickle protocol: 5```
 when trying to `dvc pull` or `dvc add` using `dvc>=2` in a directory where I still had the old cache generated using `dvc<2`. But for some reason I cannot reproduce the error any more! 🤷  Anyway, should you see that error, try to `rm -rf .dvc/cache` and then re-populate your cache with a `dvc pull` (or `dvc fetch`)—this worked for me.
 2. The new `dvc.lock` format is only slightly different but breaks backward compatibility. Also, `git diff` is confused by the new format, so please see "How to test" below to see how to visualize the changes.
3. Here are some highlights of the new elemnents introduced in `dvc 2.0`.
    1. [**ML experiments**](https://dvc.org/blog/dvc-2-0-release#lightweight-ml-experiments) (beta, may change) lets you try various hyper-parameter configurations in a lightweight way, without having to create a `git branch` for each of them.
    2. [ **`dvc run` deprecated in favor of `dvc stage add`**](https://dvc.org/blog/dvc-2-0-release#ml-model-checkpoints-versioning).
    3. [**Better metrics log and viz**](https://dvc.org/blog/dvc-2-0-release#metrics-logging).


## How to test?
1. To see what _really_ changed in any of the two `dvc.lock` files, other than the new format, please run the following command.
```bash
cp dvc.lock dvc_new.lock
git checkout master -- dvc.lock
vimdiff dvc.lock <(sed "s/^..//" dvc_new.lock | tail -n +3 | cat)  # fix formatting to allow comparison!
```
2. Follow the same steps of "How To Test" from #265, just replacing `v0.1.0` with `v0.1.1`.
## Checklist

  (if it is not the case, please create an issue first).
  (if needed)
  (if needed)
